lib/Page.js

```diff
@@ -801,11 +801,27 @@ class Page extends EventEmitter {
   }
 
   /**
-   * @param {?string} mediaType
+   * @param {?string} type
    */
-  async emulateMedia(mediaType) {
-    assert(mediaType === 'screen' || mediaType === 'print' || mediaType === null, 'Unsupported media type: ' + mediaType);
-    await this._client.send('Emulation.setEmulatedMedia', {media: mediaType || ''});
+  async emulateMediaType(type) {
+    assert(type === 'screen' || type === 'print' || type === null, 'Unsupported media type: ' + type);
+    await this._client.send('Emulation.setEmulatedMedia', {media: type || ''});
+  }
+
+  /**
+   * @param {?Array<MediaFeature>} features
+   */
+  async emulateMediaFeatures(features) {
+    if (features === null)
+      await this._client.send('Emulation.setEmulatedMedia', {features: null});
+    if (Array.isArray(features)) {
+      features.every(mediaFeature => {
+        const name = mediaFeature.name;
+        assert(/^prefers-(?:color-scheme|reduced-motion)$/.test(name), 'Unsupported media feature: ' + name);
```

Comment:
super-nit: I had to lookup what "?:" does. Do we really care if the match is captured?

Reply:
> Do we really care if the match is captured?

There's some overhead to capturing groups of the form `(pattern)`. In this case we don't need to capture at all, so we use a non-capturing group of the form `(?pattern)` instead.

Another approach here would be to maintain a set of supported media features, i.e.

```js
const SUPPORTED_MEDIA_FEATURES = new Set([
  'prefers-color-scheme',
  'prefers-reduced-motion',
]);
```

...and then check that instead of using a regular expression. This initially felt like overkill to me, because we currently only support two CSS media features, but maybe it's more readable / maintainable going forward. WDYT?

Reply:
Like I said, super-nit. I am fine with the regex as is. It is just some cognitive overhead for a tiny performance gain.